### PR TITLE
tracker2/users immer

### PR DESCRIPTION
**Pull request by chrisnojima**
- [x] immerize tracker2
- [x] immerize users (a dep of tracker2)
- [x] fix up connects of various screens i ran into that used users/tracker2
- [x] updated tracker2 to use `undefined` vs `null` more
- [x] added some types to the serialize instead of them being any
- [x] after release

### CODE REVIEW

**Comment by chrisnojima** [comment]
@keybase/react-hackers bump

**Comment by chrisnojima** [comment]
@keybase/react-hackers bump

File: shared/chat/header.desktop.tsx [hunk 1]
```diff
@@ -276,8 +271,8 @@ const Connected = Container.connect(
           : meta.teamType === 'small'
           ? meta.teamname
           : null,
-      desc,
-      fullName,
+      desc: stateProps.desc,
+      fullName: stateProps.fullName,
```
[the remark this answers is not in the document]
Comment (chrisnojima) [reply shared/chat/header.desktop.tsx]: before the entire userInfo map was passed, so any changign user anywhere would trigger a change. instead its just doing fast stuff and pulling out what it needs

File: shared/constants/tracker2.tsx [hunk 2]
```diff
@@ -128,32 +112,31 @@ export const rpcRowStateToAssertionState = (state: RPCTypes.Identify3RowState):
...
+  timestamp: row.ctime,
+  type: row.key,
+  value: row.value,
+})
 
 export const rpcSuggestionToAssertion = (s: RPCTypes.ProofSuggestion): Types.Assertion => {
   const ourKey = s.key === 'web' ? 'dnsOrGenericWebSite' : s.key
-  return makeAssertion({
+  return {
+    ...makeAssertion(),
```
[the remark this answers is not in the document]
Comment (chrisnojima) [reply shared/constants/tracker2.tsx]: just keeping what it did before

[the remark this answers is not in the document]
Comment (chrisnojima) [reply shared/constants/tracker2.tsx]: updated

File: shared/reducers/tracker2.tsx [hunk 3]
```diff
@@ -1,155 +1,149 @@
...
+  return Constants.guiIDToUsername(state, guiID)
+}
 
-      const reason =
-        action.payload.reason ||
-        (action.payload.result === 'broken' &&
-          `Some of ${username}'s proofs have changed since you last followed them.`)
+const cloneMapGetItem = (state: Types.State, username: string) => {
+  const usernameToDetails = new Map(state.usernameToDetails)
+  const old = usernameToDetails.get(username) || Constants.makeDetails()
```
[the remark this answers is not in the document]
Comment (chrisnojima) [reply shared/reducers/tracker2.tsx]: updated

File: shared/reducers/tracker2.tsx [hunk 4]
```diff
@@ -139,7 +139,7 @@ export default Container.makeReducer<
   [Tracker2Gen.loadedNonUserProfile]: (draftState, action) => {
     const {assertion, ...rest} = action.payload
     const usernameToNonUserDetails = new Map(draftState.usernameToNonUserDetails)
-    const old = usernameToNonUserDetails.get(assertion) || Constants.makeNonUserDetails()
+    const old = usernameToNonUserDetails.get(assertion) || Constants.noNonUserDetails()
```
Comment (songgao) [inline shared/reducers/tracker2.tsx]: 🔥 `()`

**Pull request merged**